use-cases/client-server-msgs-perf/svmsg_file_server.c

```diff
@@ -190,16 +252,49 @@ main(int argc, char *argv[])
             // Implement and record increment.
             clientp->last_mtype = REQ_MT_INCR;
             resp.counter = ++clientp->client_ctr;
+            clientp->num_ops++;
+
+            if (clock_gettime(one_ns_res_clock_id, &ts1)) {     // Timing ends
+                errExit("clock_gettime-ts1");
+            }
+
+            nsec0 = timespec_to_ns(&ts0);
+            nsec1 = timespec_to_ns(&ts1);
+            elapsed_ns = (nsec1 - nsec0);
+            clientp->cumu_thread_cpu_time_ns += elapsed_ns;
+
 #if L3_ENABLED
-            l3_log_simple("Server msg: ClientID=%d, Increment=%" PRIu64, resp.clientId, resp.counter);
+            // Record time-consumed, clarifying semantic of elapsed time.
+            l3_log_simple("Server msg: Increment: ClientID=%d, "
+
+#if __APPLE__
+                          "Thread-CPU-"
+#else
+                          "Elapsed real-"
+#endif  // __APPLE__
+                          "time=%" PRIu64 " ns.", resp.clientId, elapsed_ns);
 #endif // L3_ENABLED
             break;
 
           case REQ_MT_EXIT:
+
+            clientp = &ActiveClients[req.client_idx];
             // One client has informed that it has exited
             NumActiveClients--;
-            printf("Server: Client ID=%d exited. # active clients=%d\n",
-                   req.clientId, NumActiveClients);
+
+#if __APPLE__
```

Comment:
minor: you could do this once and use above.

Reply:
@gregthelaw - This code fragment is now obsolete. 

I had tried to get this working for Apple, hence, this messy code.

As I have backed-out support for Mac/OSX for now, all this has been obsoleted by a later commit.